front/src/containers/PostSurveyForm.js

```diff
@@ -117,19 +117,31 @@ class PostSurveyForm extends React.Component {
 		const numRounds = batch.numRounds;
 		const allRounds = Array.from(Array(numRounds).keys());
 		const roundPairs = batch.roundPairs;
-		const singleUserRounds = roundPairs.map(x => x.pair).map(x => x[0].roundNumber);
+
 		let nonExpRounds = allRounds;
-		if (singleUserRounds && singleUserRounds.length) {
-			nonExpRounds = allRounds.filter(x => singleUserRounds.indexOf(x) === -1);
-		}
-		let result;
-		let supposedResult = nonExpRounds.filter(x => expRounds.indexOf(x) < 0 && worstRounds.indexOf(x) < 0); // sometimes supposedResult.length can be less than 2
-		if (supposedResult && supposedResult.length >= 2) {
-			result = supposedResult;
-		} else {
-			result = nonExpRounds;
+		//Attempt to leverage batch variables
+		if(batch.dynamicTeamSize && batch.dynamicOptions){ //Team First [n,1,n,1]
+			//nonExpRounds = [0,2]
+			nonExpRounds = []
+			for (var i = 0; i < batch.numRounds; i++){
+				if(i%2==0){
+					nonExpRounds.push(i)	
+				}
```

Comment:
For safety, I tend to use a filter argument, over a loop, e.g., 
```suggestion
			nonExpRounds = nonExpRounds.filter((element, index) => index%2 === 0);
```

Note that the [spec for filter](https://www.w3schools.com/jsref/jsref_filter.asp) passes other things after element, e.g., the index: `array.filter(function(currentValue, index, arr), thisValue)`

Reply:
Also, use `===` where possible because it improves safety of the comparison ([see in the reference](https://www.w3schools.com/js/js_comparisons.asp)).

Reply:
In both this and the next one, my suggestion might not update the code accurately, because its across several lines, and I'm not sure if GitHub does that correctly.

Reply:
I will test your suggestion locally prior to pushing it!

Reply:
tested and made change!